WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormRemoteActionUseCase.swift

```diff
@@ -205,4 +258,89 @@ private extension ProductFormRemoteActionUseCase {
         }
         stores.dispatch(passwordUpdateAction)
     }
+
+    func duplicateVariations(_ variationIDs: [Int64],
+                             from oldProductID: Int64,
+                             to newProduct: EditableProductModel,
+                             onCompletion: @escaping (Result<EditableProductModel, ProductUpdateError>) -> Void) {
+        Task { [weak self] in
+            guard let self = self else { return }
+            for id in variationIDs {
+                guard let variation = await self.retrieveProductVariation(id: id, siteID: newProduct.siteID, productID: oldProductID) else {
+                    continue
+                }
+                let newVariation = CreateProductVariation(regularPrice: variation.regularPrice ?? "", attributes: variation.attributes)
+                await self.duplicateProductVariation(newVariation, parent: newProduct)
+            }
+
+            let updatedProduct: EditableProductModel = await {
+                guard let productModel = await retrieveProduct(id: newProduct.productID, siteID: newProduct.siteID) else {
+                    return newProduct
+                }
+                return EditableProductModel(product: productModel)
+            }()
+
+            await MainActor.run {
+                onCompletion(.success(updatedProduct))
+            }
+        }
+    }
+
+    func retrieveProduct(id: Int64, siteID: Int64) async -> Product? {
+        await withCheckedContinuation { [weak self] continuation in
+            guard let self = self else { return }
+            let action = ProductAction.retrieveProduct(siteID: siteID, productID: id) { result in
+                switch result {
+                case .success(let product):
+                    continuation.resume(returning: product)
+                case .failure:
+                    continuation.resume(returning: nil)
+                }
+            }
+            DispatchQueue.main.async { [weak self] in
+                self?.stores.dispatch(action)
+            }
+        } as Product?
```

Comment:
Yes, it compiles for me too, but since our CI is still running with Xcode 13 I had to add these to make it compiles. There are a few other places in the app that we're doing this - we'll need to remove them when the CI is upgraded to Xcode 14.